src/app/_components/week/weekInfo.tsx

```
@@ -4,10 +4,17 @@ import Subtitle from "../subtitle";
 import { auth } from "~/server/auth";
 
 const WeekInfo = async ({ id }: { id: number }) => {
-  const week = await api.week.getWeek(id);
   const session = await auth();
   const userId = session?.user?.id;
 
+  await api.leetcode.checkNewCompletions({
```

could we avoid making the call to the backend if we know that there is no user session?

---

You can check something like 
```typescript
    if (!session?.user) {
        return (
            <div>
                Unauthorized
            </div>
        )
    }
```

But adding

---

Would it be better to throw an error (which I think is what happens in weekly-problems?), or should it stay as a simple message? 

I was thinking something like 
```ts
if (!session?.user) {
    throw new Error("TRPCError: UNAUTHORIZED");
  }
```
but I'd like some feedback before pushing it

---

I would say the 'Unauthorized' message is a better choice, as this is not an error but adequate behavior. Perhaps we could add standard 404 or 401 pages later